### ISSUE TITLE
RFE: an interpreted language as an intermediate language to speed up testing

### ISSUE THREAD

**Issue by danbst** [vmware/differential-datalog#557]
Rust IM language is quite slow on recompiles, which is frustrating. Maybe it's possible to build template in an interpreted language (say, Python)?

This would be fine for newcomers/tutorial/CLI cases. The Rust IM language can be used for release/embedded builds though.

I understand this is quite a lot of duplication, but maybe it can be reduced with FFI to Rust non-generated objects.

**Comment by ryzhyk** [comment]
Yep, this is something we've been talking about for a while. You're right, much of the template code, in particular everything inside the `differential_datalog` crate is static and can be reused in the interpreted version. Most of the Haskell compiler logic can likewise be reused. 

We just need to find the time to build (and maintain) this, which might take a while :)

**Comment by ryzhyk** [comment]
PS. As mentioned in my other message, compilation times are much more tolerable now that we are taking advantage of incremental compilation, which is another reason this is not at the top of our todo list.

**Comment by RAbraham** [comment]
as a newbie to Rust, could you elaborate on what kind of commands you normally run for incremental compilation? Thanks.

**Comment by RAbraham** [comment]
thanks, I'll do so.
